cockatrice/src/tab_server.cpp

```diff
@@ -139,8 +139,17 @@ void RoomSelector::joinClicked()
 void RoomSelector::joinFinished(const Response &r, const CommandContainer & /*commandContainer*/, const QVariant &extraData)
 {
     switch (r.response_code()) {
-        case Response::RespOk: break;
-        case Response::RespUserLevelTooLow: QMessageBox::critical(this, tr("Error"), tr("You do not have the proper permission to join this room.")); return;
+        case Response::RespOk:
+            break;
+        case Response::RespNameNotFound:
+            QMessageBox::critical(this, tr("Error"), tr("Failed to join the room: it doesn't exists on the server."));
+            return;
+        case Response::RespContextError:
+            QMessageBox::critical(this, tr("Error"), tr("You are already inside the room."));
```

Comment:
Now that the logic flow is changed, ignore my previous comment. As far as I can tell the only way this is reached is if the server thinks we are in the room but we are not rendering a room tab - which should not happen.

Assuming my assessment is correct (please tell me if I'm wrong), then I think we should change the message here to something like `The server thinks you are in the room but Cockatrice is unable to display it. Try restarting Cockatrice`